CMakeLists.txt

```diff
@@ -1,52 +1,33 @@
 cmake_minimum_required (VERSION 2.8.0)
 project (ibmdb2i)
-set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-brtl,-lc,-blibpath:/QOpenSys/pkgs/lib:/QOpenSys/usr/lib:/usr/lib,-bbigtoc,-lpthread,-liconv")
-set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-brtl,-lc,-blibpath:/QOpenSys/pkgs/lib:/QOpenSys/usr/lib:/usr/lib,-bbigtoc,-lpthread,-liconv")
-set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -Wl,-brtl,-lc,-blibpath:/QOpenSys/pkgs/lib:/QOpenSys/usr/lib:/usr/lib,-bbigtoc,-lpthread,-liconv")
-# message("-- ibmdb2i CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS}")
-# message("-- ibmdb2i CMAKE_CXX_FLAGS_DEBUG ${CMAKE_CXX_FLAGS_DEBUG}")
-# message("-- ibmdb2i CMAKE_BUILD_TYPE ${CMAKE_BUILD_TYPE}")
 include_directories(BEFORE special/include)
 include_directories (
   ../../include
-  ../../sql 
+  ../../sql
   ../../pcre)
-SET(IBMDB2I_SOURCES db2i_blobCollection.cc
-db2i_misc.cc
-db2i_charsetSupport.cc
-db2i_collationSupport.cc
-db2i_constraints.cc
-db2i_conversion.cc
-db2i_errors.cc
-db2i_file.cc
-db2i_ileBridge.cc
-db2i_ioBuffers.cc
-db2i_myconv.cc
-db2i_rir.cc
-db2i_sqlStatementStream.cc
-db2i_global.cc
-db2i_safeString.cc
-ha_ibmdb2i.cc)
-# LINK_INTERFACE_LIBRARIES
-set(PASELIB_LIBRARIES
-    -L/QOpenSys/usr/lib
-    pthread
-    c
-    iconv)
-# STATIC_ONLY  (MANDATORY) 
-# (MANDATORY)  -- affects sql/sql_builtin.c (gen file on cmake, build-xxx.sh) 
-# -- links ibmdb2i into mysqld (daemon)
-# -- INSTALL SONAME 'ha_ibmdb2i'; (no longer required)
-# MYSQL_ADD_PLUGIN(ibmdb2i ${IBMDB2I_SOURCES} 
-#  STORAGE_ENGINE STATIC_ONLY MANDATORY
-#  LINK_LIBRARIES ${PASELIB_LIBRARIES})
 
-# MODULE_ONLY
-# -- INSTALL SONAME 'ha_ibmdb2i';
-MYSQL_ADD_PLUGIN(ibmdb2i ${IBMDB2I_SOURCES} 
- STORAGE_ENGINE MODULE_ONLY
- LINK_LIBRARIES ${PASELIB_LIBRARIES})
+SET(IBMDB2I_SOURCES
+  db2i_blobCollection.cc
+  db2i_misc.cc
+  db2i_charsetSupport.cc
+  db2i_collationSupport.cc
+  db2i_constraints.cc
+  db2i_conversion.cc
+  db2i_errors.cc
+  db2i_file.cc
+  db2i_ileBridge.cc
+  db2i_ioBuffers.cc
+  db2i_myconv.cc
+  db2i_rir.cc
+  db2i_sqlStatementStream.cc
+  db2i_global.cc
+  db2i_safeString.cc
+  ha_ibmdb2i.cc
+)
 
-# Dave add -malign-power -malign-natural (below)
-# Our power engine will run better. Also fewer gcc odd boundary over runs.
-target_compile_options(ibmdb2i PRIVATE -malign-power -malign-natural -ggdb -O0)
+MYSQL_ADD_PLUGIN(ibmdb2i
+                ${IBMDB2I_SOURCES}
+                STORAGE_ENGINE MODULE_ONLY
+                LINK_LIBRARIES /QOpenSys/usr/lib/libiconv.a)
+
+target_compile_options(ibmdb2i PRIVATE -O0)
```

Comment:
`-malign-power` and `-malign-natural`, I assume are not needed with the current code structure. 

Is the removal of `-ggdb` intentional?

Reply:
Yes we already pass `-g` for RELWITHDEBINFO or `-ggdb` DEBUG to CFLAGS during the cmake build. We can might just want to pass `-ggdb` always instead of `-g` with RELWITHDEBINFO in the mariadb spec file.

Reply:
The only point of `-malign-natural` is to force the long double field in the ILEpointer structure to be 16-byte aligned (and thus, the struct itself is 16-byte aligned), _but_ that only works if you _also_ pass `-mabi=ieeelongdouble` to cause the long double to a be a 128-bit type instead of a pair of 64-bit types (this is equivalent of -qldbl128 -qalign=natural in XLC)

So the flags as-is did nothing. We could adjust them, but instead it's better to use `__attribute__((aligned(16))` to _explicitly_ set the alignment than to mess with long doubles as a roundabout way to do the same thing.